pkg/webhook/admission/validator.go

```diff
@@ -85,33 +100,26 @@ func (h *validatingHandler) Handle(ctx context.Context, req Request) Response {
 			return Errored(http.StatusBadRequest, err)
 		}
 
-		err = obj.ValidateUpdate(oldObj)
-		if err != nil {
-			var apiStatus apierrors.APIStatus
-			if goerrors.As(err, &apiStatus) {
-				return validationResponseFromStatus(false, apiStatus.Status())
-			}
-			return Denied(err.Error())
-		}
-	}
-
-	if req.Operation == v1.Delete {
+		warnings, err = obj.ValidateUpdate(oldObj)
+	case v1.Delete:
 		// In reference to PR: https://github.com/kubernetes/kubernetes/pull/76346
 		// OldObject contains the object being deleted
-		err := h.decoder.DecodeRaw(req.OldObject, obj)
+		err = h.decoder.DecodeRaw(req.OldObject, obj)
 		if err != nil {
 			return Errored(http.StatusBadRequest, err)
 		}
 
-		err = obj.ValidateDelete()
-		if err != nil {
-			var apiStatus apierrors.APIStatus
-			if goerrors.As(err, &apiStatus) {
-				return validationResponseFromStatus(false, apiStatus.Status())
-			}
-			return Denied(err.Error())
-		}
+		warnings, err = obj.ValidateDelete()
+	default:
+		return Errored(http.StatusBadRequest, fmt.Errorf("unknown operation %q", req.Operation))
 	}
 
-	return Allowed("")
+	if err != nil {
+		var apiStatus apierrors.APIStatus
+		if errors.As(err, &apiStatus) {
+			return validationResponseFromStatus(false, apiStatus.Status())
```

Comment:
```suggestion
			return validationResponseFromStatus(false, apiStatus.Status()).WithWarnings(warnings...)
```
Should we also return the warnings here?

I'm not sure why we should ignore the warnings specifically for an APIStatus error 

same for validator_custom of course

Reply:
cc @alvaroaleman @vincepri

Reply:
not sure +1

But it seems also return the warnings make more sense because developers could just not use `warnings` with `APIStatus`.

Waiting to further comments.

Reply:
Yeah exactly. Looking at how the admission response is constructed I'm not sure if there's a reason why APIStatus errors specifically should omit warnings

Reply:
It doesn't seem a big issue to return warnings in every case as well?

Reply:
Agree!
(I think the APIStatus case is the only one where we don't do it right now)

Reply:
Updated! PTAL :heart:

/cc @vincepri @sbueringer

Reply:
Thx!